### PR TITLE
do - Add Course Form with Storybook and Tests

### LINKED ISSUE #21: Issue F1: fixtures, form, table for Courses - Form component
- [ ] There are CourseForm components with storybook entries and tests

### PR DESCRIPTION
**Pull request by darylou**
This PR adds course form with corresponding storybook and tests

[Github Storybook Link](https://ucsb-cs156-f23.github.io/proj-organic-f23-6pm-4/prs/25/storybook/?path=/story/components-courses-coursesform--create)

View:
Screenshot 2023-11-27 at 11 48 21 AM

Closes #21

### CODE REVIEW

**Review by chelseastoner — APPROVED** [review]
LGTM!

File: frontend/src/main/components/Courses/CoursesForm.js [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+                            {errors.term && 'Term is required. '}
+                        </Form.Control.Feedback>
+                    </Form.Group>
+                </Col>
+            </Row>
+
+            <Row>
+                <Col>
+                    <Form.Group className="mb-3" >
+                        <Form.Label htmlFor="start">Start(iso format)</Form.Label>
```
Comment (dayvidwang) [inline frontend/src/main/components/Courses/CoursesForm.js:98]: do you need to specify iso format when using a date picker?

File: frontend/src/main/components/Courses/CoursesForm.js [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+                            {...register("start", { required: true, pattern: isodate_regex })}
+                        />
+                        <Form.Control.Feedback type="invalid">
+                            {errors.start && 'Start date is required. '}
+                        </Form.Control.Feedback>
+                    </Form.Group>
+                </Col>
+                <Col>
+                    <Form.Group className="mb-3" >
+                        <Form.Label htmlFor="end">End(iso format)</Form.Label>
```
Comment (dayvidwang) [inline frontend/src/main/components/Courses/CoursesForm.js:113]: do you need to specify iso format when using a date picker? (same as above)

File: frontend/src/tests/components/Courses/CoursesForm.test.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+                <CoursesForm initialContents={coursesFixtures.oneCourse} />
+            </Router>
+        );
+        await screen.findByTestId(/CoursesForm-id/);
+        expect(screen.getByText(/Id/)).toBeInTheDocument();
+        expect(screen.getByTestId(/CoursesForm-id/)).toHaveValue("1");
+    });
+
+
+    test("Correct Error messsages on missing input", async () => {
```
Comment (dayvidwang) [inline frontend/src/tests/components/Courses/CoursesForm.test.js:41]: (nitpick) **messages**

File: frontend/src/tests/components/Courses/CoursesForm.test.js [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+        fireEvent.change(schoolField, { target: { value: 'ucsb' } });
+        fireEvent.change(termField, { target: { value: 'f23' } });
+        fireEvent.change(startField, { target: { value: '2022-01-02T12:00' } });
+        fireEvent.change(endField, { target: { value: '2022-02-02T12:00' } });
+        fireEvent.change(githubOrgField, { target: { value: 'cs156-f23'}})
+        fireEvent.click(submitButton);
+
+        await waitFor(() => expect(mockSubmitAction).toHaveBeenCalled());
+
+        expect(screen.queryByText(/Start date is required./)).not.toBeInTheDocument();
```
Comment (dayvidwang) [inline frontend/src/tests/components/Courses/CoursesForm.test.js:92]: is there a reason you use getByText above and queryByText here?

File: frontend/src/tests/components/Courses/CoursesForm.test.js [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+        await screen.findByText(/Name is required./);
+        expect(screen.getByText(/Name is required/)).toBeInTheDocument();
+        expect(screen.getByText(/School is required./)).toBeInTheDocument();
+        expect(screen.getByText(/Term is required./)).toBeInTheDocument();
+        expect(screen.getByText(/Start date is required./)).toBeInTheDocument();
+        expect(screen.getByText(/End date is required./)).toBeInTheDocument();
+        expect(screen.getByText(/GithubOrg is required./)).toBeInTheDocument();
+    });
+
+    test("No Error messsages on good input", async () => {
```
Comment (dayvidwang) [inline frontend/src/tests/components/Courses/CoursesForm.test.js:62]: (nitpick) **messages**

**Review by dayvidwang — APPROVED** [review]

File: frontend/src/tests/components/Courses/CoursesForm.test.js (hunk 4 above)
Comment (darylou) [reply frontend/src/tests/components/Courses/CoursesForm.test.js:92]: Using getByText when it isn't in the document will lead to an error. This test is checking if that text is not in the document.

File: frontend/src/main/components/Courses/CoursesForm.js (hunk 1 above)
Comment (darylou) [reply frontend/src/main/components/Courses/CoursesForm.js:98]: team03 followed a similar example so I went with that. I don't think so though.

File: frontend/src/main/components/Courses/CoursesForm.js (hunk 2 above)
Comment (darylou) [reply frontend/src/main/components/Courses/CoursesForm.js:113]: team03 followed a similar example so I went with that. I don't think so though.

File: frontend/src/tests/components/Courses/CoursesForm.test.js (hunk 3 above)
Comment (darylou) [reply frontend/src/tests/components/Courses/CoursesForm.test.js:41]: Ill probably fix this in another PR but good catch!

File: frontend/src/tests/components/Courses/CoursesForm.test.js (hunk 5 above)
Comment (darylou) [reply frontend/src/tests/components/Courses/CoursesForm.test.js:62]: Ill probably fix this in another PR but good catch!

File: frontend/src/main/components/Courses/CoursesForm.js (hunk 1 above)
Comment (pconrad) [reply frontend/src/main/components/Courses/CoursesForm.js:98]: It's true: you don't really need this.  I've asked other teams to take it out, but for sake of moving things along, I'll go ahead and let it slide.  But you might want to do a 5 pt refactoring PR later to remove those checks.

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**